tap_postgres/tap.py

```diff
@@ -33,7 +80,82 @@ def connector(self) -> PostgresConnector:
         Connector is a singleton (one instance is used by the Tap and Streams).
 
         """
-        return PostgresConnector(dict(self.config))
+        # We mutate this url to use the ssh tunnel if enabled
+        self.url = make_url(self.config["sqlalchemy_url"])
+
+        if self.config["ssh_tunnel.enable"]:
+            self.ssh_tunnel_connect()
+        return PostgresConnector(
+            config=dict(self.config),
+            sqlalchemy_url=str(
+                self.url
+            ),  # Sometimes we mutuate the url to use the ssh tunnel
+        )
+
+    def guess_key_type(self, key_data):
+        """We are duplciating some logic from the ssh_tunnel package here,
+        we could try to use their function instead."""
+        for key_class in [
+            paramiko.RSAKey,
+            paramiko.DSSKey,
+            paramiko.ECDSAKey,
+            paramiko.Ed25519Key,
+        ]:
+            try:
+                key = key_class.from_private_key(io.StringIO(key_data))
+                return key
+            except paramiko.SSHException:
+                continue
+        raise ValueError("Could not determine the key type.")
+
+    def ssh_tunnel_connect(self) -> None:
```

Comment:
Wdyt of returning a new URL instead of overwriting an attribute here? E.g.

```python
def ssh_tunnel_connect(self, url) -> None:
    ...
    return url.set(
        host=self.ssh_tunnel.local_bind_host,
        port=self.ssh_tunnel.local_bind_port,
    )
```

That way you don't need a `.url` attribute with an unclear lifecycle.